### DISCUSSION TITLE
How can we skip a vulnerability to be reported if fixed version information is not available.

### DISCUSSION THREAD

**Discussion by chandrash3kh4r** [aquasecurity/trivy#1687] [Q&A] [answered]
I am running trivy scan on a docker image using below command.
trivy -f json -o trivy_high_critical_results.json --severity CRITICAL,HIGH --exit-code 1 ${DOCKER_IMAGE_NAME}:${TAG}

We want this command to not fail if the report CRITICAL or HIGH vulnerability does not have a fixed version available. 
Is it possible? Any suggestions?

Thanks in advance.

**Comment by knqyf263** [comment]
@DmitriyLewen Can you help him?

**Comment by DmitriyLewen** [comment, marked as answer]
Hello @chandrash3kh4r 
Thanks for your interest in trivy.
You can use [--ignore-unfixed](https://github.com/aquasecurity/trivy/blob/main/docs/getting-started/cli/image.md) flag.
This flag excludes vulnerabilities for which there are no fixed versions. 
Is it ok for you?

**Reply by chandrash3kh4r** [reply]
Thanks a lot @DmitriyLewen it worked.

**Comment by kmganna** [comment]
--ignore-unfixed option is really helpful but during trivy report upload, ignore-unfixed is unable to ignore the severity in the git actions. It is giving all the severity issues.

**Reply by knqyf263** [reply]
It should work. Please share your setting.

**Reply by kmganna** [reply]
Let me explain the requirement.

I am running the trivy vulnerability scan on the docker image in the git action pipeline. 

Below is the code.
` - name: Run Trivy vulnerability scanner on a container
     uses: aquasecurity/trivy-action@9ab158e8597f3b310480b9a69402b419bc03dbd5 
        with:
          image-ref: '${{ inputs.registry }}/${{ inputs.repo_name }}:${{ inputs.image_version }}'
          format: 'sarif'
          severity: 'CRITICAL,HIGH,MEDIUM'  
          output: 'trivy-container-results.sarif'
          ignore-unfixed: true
          timeout: ${{ inputs.timeout }}`
          
The above code is working and giving the results based on the severity of the application and it is ignoring unfixed OS fixes(HIGH, MEDIUM and CRITICAL) and LOW fixes as well. Now if I make ignore-unfixed: false then it is giving all the ignored OS fixes along with "LOW severity".
    
**Requirement is:**
    If I use ignore-unfixed: false and there is a severity CRITICAL, HIGH, and MEDIUM. The output should be application issues and OS issues in the output file based on severity. It should not be LOW and other issues.
    
Please suggest how to achieve it.

**Reply by knqyf263** [reply]
@AndreyLevchenko Would you look into it?

**Reply by AndreyLevchenko** [reply]
yes, let me check if I can reproduce it

**Reply by AndreyLevchenko** [reply]
@knqyf263 
This is trivy action specific behavior 
https://github.com/aquasecurity/trivy-action/blob/master/entrypoint.sh#L178-L184
@simar7 may have better explanation why it's working this way

**Reply by knqyf263** [reply]
Thanks. Seems like a bug in trivy-aciton.
@kmganna Could you raise an issue in trivy-action?

**Reply by kmganna** [reply]
Thanks @knqyf263 and @AndreyLevchenko 

I have raised an issue in the trivy-action
https://github.com/aquasecurity/trivy-action/issues/192
